homeassistant/components/sensor/dht.py

```diff
@@ -39,9 +39,8 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         "DHT22": Adafruit_DHT.DHT22,
         "AM2302": Adafruit_DHT.AM2302
     }
-    sensor = available_sensors[config['sensor']]
-
-    pin = config['pin']
+    sensor = available_sensors.get(config.get('sensor'))
+    pin = config.get('pin')
```

Comment:
I hate magic strings so I'd agree all of em should be constants.

On Aug 24, 2016 10:39 PM, "Roi Dayan" <notifications@github.com> wrote:

> In homeassistant/components/sensor/dht.py
> <https://github.com/home-assistant/home-assistant/pull/2956#discussion_r76179199>
> :
>
> > @@ -39,9 +39,8 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
> >          "DHT22": Adafruit_DHT.DHT22,
> >          "AM2302": Adafruit_DHT.AM2302
> >      }
> > -    sensor = available_sensors[config['sensor']]
> > -
> > -    pin = config['pin']
> > +    sensor = available_sensors.get(config.get('sensor'))
> > +    pin = config.get('pin')
>
> do you mean PIN = 'pin' ? it seems redundant. what about sensor,
> temperature, humidity ?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/2956/files/989e873b79f41335b3c8310e97d992417ccab20b#r76179199>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC2fZY8qyfhmn9dhBssLFZ3SKVLsAa0Sks5qjRyQgaJpZM4Jrx53>
> .
>